Classes/Controller/SearchController.php

```diff
@@ -393,6 +393,13 @@ private function createQuery () {
 		$query = $this->solr->createSelect();
 		$this->addTypoScriptFilters($query);
 
+        if(count($this->settings['shards'])) {
```

Comment:
I was thinking about that too. Decided against it to keep query creation code together. But after another thought encapsulating inside a function is better. Will do a new PR. 

Thx for reviewing.